src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    /**
+     * Returns the Double value of numberString
+     */
+    private Double parseDouble(String numberString) throws BadRequestException {
+      Double number = null;
+      try {
+        number = Double.parseDouble(numberString);
+      } catch (Exception e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return number;
+    }
+
+    /**
+     * Returns LocalTime object from timeString
+     */
+    private LocalTime parseTime(String timeString) throws BadRequestException {
+      LocalTime time = null;
+      DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm");
+      try {
+        time = LocalTime.parse(timeString, formatter);
+      } catch (DateTimeParseException e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return time;
+    }
+
+    private int getNumberOfSeconds(LocalTime time) {
+      return time.getHour() * 3600 + time.getMinute() * 60;
+    }
+  }
+
+  private CreateSlotManager slotRequest;
+  private String userId;
+  
+  /** 
+   * Creates a delivery slot request given the delivery day, the timezone offset in minutes, start
+   * and end times of the delivery, the maximum number of stops and user's id.
+   */
+  public void createSlot(String deliveryDay, String timezoneOffsetMinutes, String startTime, String endTime, String maxStops, String userId) throws BadRequestException {
+    this.slotRequest = new CreateSlotManager(deliveryDay, timezoneOffsetMinutes, startTime, endTime, maxStops);
+    this.userId = userId;
+  }
+
+  public void setSlotStartingPoint(String address) throws ApiException, InterruptedException, IOException, DataNotFoundException {
+    this.slotRequest.setCoordinatesFromAddress(address);
+  }
+  public void setSlotStartingPoint(String latitude, String longitude) throws BadRequestException {
+    this.slotRequest.setCoordinates(latitude, longitude);
+  }
+  public DeliverySlot getDeliverySlot() {
+    return slotRequest.deliverySlot;
+  }
+  public double getStartLatitude() {
+    return slotRequest.latitude;
+  }
+  public double getStartLongitude() {
+    return slotRequest.longitude;
+  }
+  public String getUserId() {
+    return userId;
+  }
+
+  public void addSlotToDatastore() {
+    DatastoreServiceFactory.getDatastoreService().put(createDeliveryRequestEntity());
+  }
+
+  /**
+   * Create a deliveryRequest Entity with properties' values from slotManager.
+   */
+  private Entity createDeliveryRequestEntity() {
+    Entity deliveryRequest = new Entity("deliveryRequest");
+
+    deliveryRequest.setProperty("deliveryDay", slotRequest.deliverySlot.getDeliveryDay());
+    deliveryRequest.setProperty("startTime", slotRequest.deliverySlot.getStartTime());
+    deliveryRequest.setProperty("endTime", slotRequest.deliverySlot.getEndTime());
+    deliveryRequest.setProperty("uid", userId);
+    deliveryRequest.setProperty("startLat", slotRequest.latitude);
+    deliveryRequest.setProperty("startLng", slotRequest.longitude);
```

Comment:
Could all of these strings go into an Enum as we did before?

Reply:
Done.